diff -urN a/featherpad/syntax.cpp b/featherpad/syntax.cpp
--- a/featherpad/syntax.cpp	
+++ b/featherpad/syntax.cpp	
@@ -87,6 +87,7 @@
             progLan = "c";
         else if (fname.endsWith (".sh") || fname.endsWith (".bashrc") || fname.endsWith (".rules")
                  || fname.endsWith (".zsh") || baseName == ".bash_profile"
+                 || fname.endsWith (".ebuild") || fname.endsWith (".eclass")
                  || baseName == ".bash_functions" || baseName == ".bash_logout"
                  || baseName == ".bash_aliases" || baseName == ".xprofile"
                  || baseName == ".profile" || baseName == ".mkshrc"
