diff -urN a/featherpad/syntax.cpp b/featherpad/syntax.cpp
--- a/featherpad/syntax.cpp	2022-10-13 19:56:42.366520584 -0500
+++ b/featherpad/syntax.cpp	2022-10-13 20:18:01.127178593 -0500
@@ -87,6 +87,7 @@
             progLan = "c";
         else if (fname.endsWith (".sh") || fname.endsWith (".bashrc") || fname.endsWith (".rules")
                  || fname.endsWith (".zsh")
+                 || fname.endsWith (".ebuild") || fname.endsWith (".eclass")
                  || baseName == ".bash_profile" || baseName == ".bash_functions"
                  || baseName == ".xprofile" || baseName == ".profile"
                  || baseName == ".bash_aliases" || baseName == ".mkshrc"
