From 8e021a46ee2d455c8c677a7eb982b56c3c408942 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Mon, 18 Feb 2019 14:09:13 +0200
Subject: dir_setlasterr: strcpy the right variable

---
 src/dir_common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/dir_common.c b/src/dir_common.c
index 234ab9b..ee80c81 100644
--- a/src/dir_common.c
+++ b/src/dir_common.c
@@ -87,7 +87,7 @@ void dir_setlasterr(const char *path) {
     lasterrl = req;
     lasterr = realloc(lasterr, lasterrl);
   }
-  strcpy(lasterr, dir_curpath);
+  strcpy(lasterr, path);
 }
 
 
-- 
cgit v1.2.1

