diff -ruN a/net/wireless/reg.c b/net/wireless/reg.c
--- a/net/wireless/reg.c	2024-10-21 17:02:59.573009991 -0400
+++ b/net/wireless/reg.c	2024-10-28 18:24:45.340878841 -0400
@@ -238,43 +238,36 @@
 static void reg_todo(struct work_struct *work);
 static DECLARE_WORK(reg_work, reg_todo);
 
-/* We keep a static world regulatory domain in case of the absence of CRDA */
 static const struct ieee80211_regdomain world_regdom = {
-	.n_reg_rules = 8,
-	.alpha2 =  "00",
+	.n_reg_rules = 9,
+	.alpha2 = "00",
 	.reg_rules = {
-		/* IEEE 802.11b/g, channels 1..11 */
-		REG_RULE(2412-10, 2462+10, 40, 6, 20, 0),
-		/* IEEE 802.11b/g, channels 12..13. */
-		REG_RULE(2467-10, 2472+10, 20, 6, 20,
-			NL80211_RRF_NO_IR | NL80211_RRF_AUTO_BW),
-		/* IEEE 802.11 channel 14 - Only JP enables
-		 * this and for 802.11b only */
-		REG_RULE(2484-10, 2484+10, 20, 6, 20,
-			NL80211_RRF_NO_IR |
-			NL80211_RRF_NO_OFDM),
-		/* IEEE 802.11a, channel 36..48 */
-		REG_RULE(5180-10, 5240+10, 80, 6, 20,
-                        NL80211_RRF_NO_IR |
-                        NL80211_RRF_AUTO_BW),
-
-		/* IEEE 802.11a, channel 52..64 - DFS required */
-		REG_RULE(5260-10, 5320+10, 80, 6, 20,
-			NL80211_RRF_NO_IR |
-			NL80211_RRF_AUTO_BW |
-			NL80211_RRF_DFS),
-
-		/* IEEE 802.11a, channel 100..144 - DFS required */
-		REG_RULE(5500-10, 5720+10, 160, 6, 20,
-			NL80211_RRF_NO_IR |
-			NL80211_RRF_DFS),
-
-		/* IEEE 802.11a, channel 149..165 */
-		REG_RULE(5745-10, 5825+10, 80, 6, 20,
-			NL80211_RRF_NO_IR),
+		/* 2.4 GHz, channels 1-11 */
+		REG_RULE(2412 - 10, 2462 + 10, 40, 8, 40, 0),
 
-		/* IEEE 802.11ad (60GHz), channels 1..3 */
-		REG_RULE(56160+2160*1-1080, 56160+2160*3+1080, 2160, 0, 0, 0),
+		/* 2.4 GHz, channels 12-13 */
+		REG_RULE(2467 - 10, 2472 + 10, 20, 8, 40, 0),
+
+		/* 2.4 GHz, channel 14 - only Japan, 802.11b only */
+		REG_RULE(2484 - 10, 2484 + 10, 20, 8, 40, 0),
+
+		/* 5 GHz, channels 36-48 */
+		REG_RULE(5180 - 10, 5240 + 10, 80, 8, 40, 0),
+
+		/* 5 GHz, channels 52-64 */
+		REG_RULE(5260 - 10, 5320 + 10, 80, 8, 40, 0),
+
+		/* 5 GHz, channels 100-144 */
+		REG_RULE(5500 - 10, 5720 + 10, 160, 8, 40, 0),
+
+		/* 5 GHz, channels 149-165 */
+		REG_RULE(5745 - 10, 5825 + 10, 80, 8, 40, 0),
+
+		/* 6 GHz, channels 1-233 */
+		REG_RULE(5955 - 10, 7115 + 10, 160, 8, 40, 0),
+
+		/* 60 GHz, channels 1-3 */
+		REG_RULE(56160 + 2160 * 1 - 1080, 56160 + 2160 * 3 + 1080, 2160, 8, 40, 0),
 	}
 };
 
