--- a/configure	
+++ b/configure
@@ -10109,7 +10109,7 @@
     else
         # It is not an error if an installed version of Zip can't be located.
         # We can use the locally distributed minizip instead
-        ZIP_PROG="../minizip${EXEEXT_FOR_BUILD}"
+        ZIP_PROG="unzip"
         ZIP_PROG_OPTIONS="-o"
         ZIP_PROG_VFSSEARCH="."
         ZIP_INSTALL_OBJS="minizip${EXEEXT_FOR_BUILD}"
