diff -urN a/basic-conf b/basic-conf
--- a/basic-conf	2015-02-13 21:57:45.000000000 -0600
+++ b/basic-conf	1969-12-31 18:00:00.000000000 -0600
@@ -1,52 +0,0 @@
-// Only use_authtok (authentication token) when using cracklib or some other module
-// that checks for passwords, or pam_krb5
-#define AUTHTOK use_authtok
-
-#if HAVE_CRACKLIB || HAVE_PASSWDQC
-# define PASSWORD_STRENGTH 1
-#endif
-
-#if HAVE_KRB5 && PASSWORD_STRENGTH
-# define KRB5_AUTHTOK AUTHTOK
-#endif
-
-#if HAVE_KRB5 || PASSWORD_STRENGTH
-# define UNIX_AUTHTOK AUTHTOK
-#else
-# define UNIX_AUTHTOK
-#endif
-
-// Define DEBUG to an empty string unless it was required by the user
-#ifndef DEBUG
-#define DEBUG
-#endif
-
-#ifndef UNIX_EXTENDED_ENCRYPTION
-#define UNIX_EXTENDED_ENCRYPTION
-#endif
-
-#ifndef LIKEAUTH
-#define LIKEAUTH
-#endif
-
-#if WANT_NULLOK
-#define NULLOK nullok
-#else
-#define NULLOK
-#endif
-
-#define KRB5_PARAMS DEBUG ignore_root try_first_pass
-
-/* By using the extended Linux-PAM syntax for this, it is possible to
-   fine-tune the Kerberos handling so that it works out of hte box on
-   most desktop systems.
-
-   What this control operation does is ignore failures and errors from
-   Kerberos (falling back on local pam_unix auth), but if it's good,
-   it'll skip over the following module (pam_unix) with an accepted
-   status.
-
-   IMPORTANT! Make sure that the only thing that comes right after
-   pam_krb5 with KRB5_CONTROL is pam_unix!
- */
-#define KRB5_CONTROL [success=1 default=ignore]
diff -urN a/.gitignore b/.gitignore
--- a/.gitignore	2015-02-13 21:57:45.000000000 -0600
+++ b/.gitignore	1969-12-31 18:00:00.000000000 -0600
@@ -1,10 +0,0 @@
-login
-passwd
-su
-system-auth
-system-login
-system-local-login
-system-remote-login
-system-services
-other
-pambase-*.tar.bz2
diff -urN a/linux-pam-conf b/linux-pam-conf
--- a/linux-pam-conf	2015-02-13 21:57:45.000000000 -0600
+++ b/linux-pam-conf	2018-09-03 09:59:14.308265040 -0500
@@ -1,3 +1,30 @@
+// Only use_authtok (authentication token) when using cracklib or some other module
+// that checks for passwords, or pam_krb5
+#if HAVE_CRACKLIB
+	# define PASSWORD_STRENGTH 1
+#endif
+
+# define UNIX_AUTHTOK
+
+// Define DEBUG to an empty string unless it was required by the user
+#ifndef DEBUG
+	#define DEBUG
+#endif
+
+#ifndef UNIX_EXTENDED_ENCRYPTION
+	#define UNIX_EXTENDED_ENCRYPTION
+#endif
+
+#ifndef LIKEAUTH
+	#define LIKEAUTH
+#endif
+
+#if WANT_NULLOK
+	#define NULLOK nullok
+#else
+	#define NULLOK
+#endif
+
 #define HAVE_LIMITS			1
 #define HAVE_ENV			1
 #define HAVE_ACCESS			1
@@ -9,23 +36,13 @@
 #define SUPPORT_NOLOGIN_AUTH		1
 
 #if !MINIMAL
-# define HAVE_MOTD			1
-# define HAVE_MAIL			1
-# define HAVE_LASTLOG                   1
-
-# if LINUX_PAM_VERSION > 0x010100 /* 1.1.0 */
-#  define TALLY_MODULE pam_tally2.so
-# else
-#  define TALLY_MODULE pam_tally.so
-# endif
-
+	# define HAVE_MOTD			1
+	# define HAVE_MAIL			1
+	# define HAVE_LASTLOG                   1
+	# define TALLY_MODULE pam_tally2.so
 #endif
 
-#if WANT_SHA512
 # define UNIX_EXTENDED_ENCRYPTION	sha512 shadow
-#else
-# define UNIX_EXTENDED_ENCRYPTION	md5 shadow
-#endif
 
 #define LIKEAUTH likeauth
 #define DEBUG_NOLOGIN
diff -urN a/login.in b/login.in
--- a/login.in	2015-02-13 21:57:45.000000000 -0600
+++ b/login.in	2018-09-03 10:00:16.723635056 -0500
@@ -1,5 +1,5 @@
 #if HAVE_SECURETTY
-auth       required	pam_securetty.so
+	auth       required	pam_securetty.so
 #endif
 auth       include	system-local-login
 
diff -urN a/Makefile b/Makefile
--- a/Makefile	2015-02-13 21:57:45.000000000 -0600
+++ b/Makefile	2018-09-03 10:01:12.007077027 -0500
@@ -14,55 +14,29 @@
 IMPLEMENTATION=linux-pam
 endif
 
-PAMFLAGS = -include $(IMPLEMENTATION)-conf -include basic-conf -DLINUX_PAM_VERSION=$(LINUX_PAM_VERSION)
+PAMFLAGS = -include $(IMPLEMENTATION)-conf -DLINUX_PAM_VERSION=$(LINUX_PAM_VERSION)
 
 ifeq "$(CRACKLIB)" "yes"
 PAMFLAGS += -DHAVE_CRACKLIB=1
 endif
 
-ifeq "$(PASSWDQC)" "yes"
-PAMFLAGS += -DHAVE_PASSWDQC=1
-endif
-
-ifeq "$(CONSOLEKIT)" "yes"
-PAMFLAGS += -DHAVE_CONSOLEKIT=1
-endif
-
 ifeq "$(SYSTEMD)" "yes"
 PAMFLAGS += -DHAVE_SYSTEMD=1
 endif
 
-ifeq "$(GNOME_KEYRING)" "yes"
-PAMFLAGS += -DHAVE_GNOME_KEYRING=1
-endif
-
 ifeq "$(SECURETTY)" "yes"
 PAMFLAGS += -DHAVE_SECURETTY=1
 endif
 
-ifeq "$(SELINUX)" "yes"
-PAMFLAGS += -DHAVE_SELINUX=1
-endif
-
 ifeq "$(MKTEMP)" "yes"
 PAMFLAGS += -DHAVE_MKTEMP=1
 endif
 
-ifeq "$(PAM_SSH)" "yes"
-PAMFLAGS += -DHAVE_PAM_SSH=1
-endif
-
-ifeq "$(KRB5)" "yes"
-PAMFLAGS += -DHAVE_KRB5=1
-endif
-
 ifeq "$(NULLOK)" "yes"
 PAMFLAGS += -DWANT_NULLOK=1
 endif
 
-ifeq "$(SHA512)" "yes"
 PAMFLAGS += -DWANT_SHA512=1
-endif
 
 ifeq "$(DEBUG)" "yes"
 PAMFLAGS += -DDEBUG=debug
diff -urN a/openpam-conf b/openpam-conf
--- a/openpam-conf	2015-02-13 21:57:45.000000000 -0600
+++ b/openpam-conf	1969-12-31 18:00:00.000000000 -0600
@@ -1,24 +0,0 @@
-#if HAVE_CRACKLIB
-# error "pam_cracklib is only supported with Linux-PAM"
-#endif
-
-// OpenPAM only provides basic modules, it's FreeBSD that provides the
-// extended modules, so check for FreeBSD building first.
-#ifdef __FreeBSD__
-# define HAVE_LOGIN_ACCESS		1
-# define SUPPORT_UNIX_SESSION		0
-
-# if defined(DEBUG)
-#  define DEBUG_NOLOGIN DEBUG
-# else
-#  define DEBUG_NOLOGIN
-# endif
-
-# define SUPPORT_NOLOGIN_ACCOUNT	1
-# define SUPPORT_NOLOGIN_AUTH		0
-
-# if defined(WANT_SHA512)
-#  error "SHA512 support is not present for FreeBSD!"
-# endif
-
-#endif /* __FreeBSD__ */
diff -urN a/passwd.in b/passwd.in
--- a/passwd.in	2015-02-13 21:57:45.000000000 -0600
+++ b/passwd.in	2018-09-03 10:02:08.146510813 -0500
@@ -1,6 +1,4 @@
-auth       sufficient   pam_rootok.so
-auth       include	system-auth
-
-account    include	system-auth
-
-password   include	system-auth
+auth		sufficient	pam_rootok.so
+auth		include		system-auth
+account		include		system-auth
+password	include		system-auth
diff -urN a/README b/README
--- a/README	2015-02-13 21:57:45.000000000 -0600
+++ b/README	1969-12-31 18:00:00.000000000 -0600
@@ -1,8 +0,0 @@
-This repository contains the PAM configuration base for Gentoo Linux
-and Gentoo FreeBSD, this mostly means the system-auth and system-login
-configuration file that provides the basic support for generical
-authentication for services, and console login (on tty or on various
-desktop managers).
-
-The Makefile manages the choice of optional features that will be
-enabled in the final file.
diff -urN a/su.in b/su.in
--- a/su.in	2015-02-13 21:57:45.000000000 -0600
+++ b/su.in	2018-09-03 10:02:20.250389014 -0500
@@ -1,11 +1,8 @@
 auth       sufficient	pam_rootok.so
 auth       required     pam_wheel.so use_uid
 auth       include		system-auth
-
 account    include		system-auth
-
 password   include		system-auth
-
 session    include		system-auth
 session    required     pam_env.so
 session    optional		pam_xauth.so
diff -urN a/system-auth.in b/system-auth.in
--- a/system-auth.in	2015-02-13 21:57:45.000000000 -0600
+++ b/system-auth.in	2018-09-03 10:08:25.934708990 -0500
@@ -1,41 +1,19 @@
-#if HAVE_ENV
-auth		required	pam_env.so DEBUG
-#endif
-#if HAVE_PAM_SSH
-auth		sufficient	pam_ssh.so
-#endif
-#if HAVE_KRB5
-auth		KRB5_CONTROL	pam_krb5.so KRB5_PARAMS
-#endif
-auth		required	pam_unix.so try_first_pass LIKEAUTH NULLOK DEBUG
-/* This is needed to make sure that the Kerberos skip-on-success won't cause a bad jump. */
-auth		optional	pam_permit.so
+auth      required  pam_unix.so     try_first_pass nullok
+auth      optional  pam_permit.so
+auth      required  pam_env.so
 
-#if HAVE_KRB5
-account		KRB5_CONTROL	pam_krb5.so KRB5_PARAMS
-#endif
-account		required	pam_unix.so DEBUG
-/* This is needed to make sure that the Kerberos skip-on-success won't cause a bad jump. */
-account		optional	pam_permit.so
+account   required  pam_unix.so
+account   optional  pam_permit.so
+account   required  pam_time.so
 
 #if HAVE_CRACKLIB
-password	required	pam_cracklib.so difok=2 minlen=8 dcredit=2 ocredit=2 retry=3 DEBUG
+	password	required	pam_cracklib.so difok=2 minlen=8 dcredit=2 ocredit=2 retry=3 DEBUG
 #endif
-#if HAVE_PASSWDQC
-password	required	pam_passwdqc.so min=8,8,8,8,8 retry=3
-#endif
-#if HAVE_KRB5
-password	KRB5_CONTROL	pam_krb5.so KRB5_PARAMS
-#endif
-password	required	pam_unix.so try_first_pass UNIX_AUTHTOK NULLOK UNIX_EXTENDED_ENCRYPTION DEBUG
-/* This is needed to make sure that the Kerberos skip-on-success won't cause a bad jump. */
-password	optional	pam_permit.so
+password  required  pam_unix.so     try_first_pass nullok sha512 shadow
+password  optional  pam_permit.so
 
-#if HAVE_PAM_SSH
-session		optional	pam_ssh.so
-#endif
-#include "system-session.inc"
+session   required  pam_limits.so
+session   required  pam_unix.so
+session   optional  pam_permit.so
 
-#if HAVE_SYSTEMD
 -session        optional        pam_systemd.so
-#endif
diff -urN a/system-login.in b/system-login.in
--- a/system-login.in	2015-02-13 21:57:45.000000000 -0600
+++ b/system-login.in	2018-09-03 10:07:34.023231382 -0500
@@ -1,62 +1,18 @@
-#if defined(TALLY_MODULE)
-auth		required	TALLY_MODULE onerr=succeed
-#endif
-#if HAVE_SHELLS
-auth		required	pam_shells.so DEBUG
-#endif
-#if SUPPORT_NOLOGIN_AUTH
-auth		required	pam_nologin.so DEBUG_NOLOGIN
-#endif
-auth		include		system-auth
-#if HAVE_GNOME_KEYRING
-auth		optional	pam_gnome_keyring.so
-#endif
+auth       required   pam_tally.so         onerr=succeed file=/var/log/faillog
+auth       required   pam_shells.so
+auth       requisite  pam_nologin.so
+auth       include    system-auth
 
-#if HAVE_ACCESS
-account		required	pam_access.so DEBUG
-#endif
-#if HAVE_LOGIN_ACCESS
-account		required	pam_login_access.so
-#endif
-#if SUPPORT_NOLOGIN_ACCOUNT
-account		required	pam_nologin.so DEBUG_NOLOGIN
-#endif
-account		include		system-auth
-#if defined(TALLY_MODULE)
-account		required	TALLY_MODULE onerr=succeed DEBUG
-#endif
+account    required   pam_access.so
+account    required   pam_nologin.so
+account    include    system-auth
 
-password	include		system-auth
-#if HAVE_GNOME_KEYRING
-password	optional	pam_gnome_keyring.so
-#endif
+password   include    system-auth
 
-#if HAVE_LOGINUID
-session         optional        pam_loginuid.so
-#endif
-#if HAVE_SELINUX
-session		required	pam_selinux.so close
-#endif
-#if HAVE_ENV
-session		required	pam_env.so DEBUG
-#endif
-#if HAVE_LASTLOG
-session		optional	pam_lastlog.so silent DEBUG
-#endif
-session		include		system-auth
-#if HAVE_CONSOLEKIT
-session		optional	pam_ck_connector.so nox11
-#endif
-#if HAVE_SELINUX
-# Note: modules that run in the user's context must come after this line.
-session		required	pam_selinux.so multiple open
-#endif
-#if HAVE_GNOME_KEYRING
-session		optional	pam_gnome_keyring.so auto_start
-#endif
-#if HAVE_MOTD
-session		optional	pam_motd.so motd=/etc/motd
-#endif
-#if HAVE_MAIL
-session		optional	pam_mail.so
-#endif
+session    optional   pam_loginuid.so
+session    optional   pam_keyinit.so       force revoke
+session    include    system-auth
+session    optional   pam_motd.so          motd=/etc/motd
+session    optional   pam_mail.so          dir=/var/spool/mail standard quiet
+-session   optional   pam_systemd.so
+session    required   pam_env.so
diff -urN a/system-services.in b/system-services.in
--- a/system-services.in	2015-02-13 21:57:45.000000000 -0600
+++ b/system-services.in	2018-09-03 10:08:02.070949132 -0500
@@ -1,8 +1,9 @@
-auth		sufficient	pam_permit.so
+auth      sufficient  pam_permit.so
 
-account		include		system-auth
+account   include     system-auth
 
-#if HAVE_LOGINUID
-session         optional        pam_loginuid.so
-#endif
-#include "system-session.inc"
+session   optional    pam_loginuid.so
+session   required    pam_limits.so
+session   required    pam_unix.so
+session   optional    pam_permit.so
+session   required    pam_env.so
diff -urN a/system-session.inc b/system-session.inc
--- a/system-session.inc	2015-02-13 21:57:45.000000000 -0600
+++ b/system-session.inc	1969-12-31 18:00:00.000000000 -0600
@@ -1,25 +0,0 @@
-#if HAVE_LIMITS
-session		required	pam_limits.so DEBUG
-#endif
-#if HAVE_ENV
-session		required	pam_env.so DEBUG
-#endif
-#if HAVE_MKTEMP
-session		optional	pam_mktemp.so
-#endif
-
-/* Only Linux-PAM supports session chain for pam_unix; but if it were
-   to not support it for whatever reason, still execute pam_krb5, with
-   sufficient level instead. */
-#if SUPPORT_UNIX_SESSION
-# if HAVE_KRB5
-session		KRB5_CONTROL	pam_krb5.so KRB5_PARAMS
-# endif
-session		required	pam_unix.so DEBUG
-#else
-# if HAVE_KRB5
-session		sufficient	pam_krb5.so KRB5_PARAMS
-# endif
-#endif
-
-session		optional	pam_permit.so
