diff --git a/grub-core/script/parser.y b/grub-core/script/parser.y
index 1b93ecc57..f8c8e5f51 100644
--- a/grub-core/script/parser.y
+++ b/grub-core/script/parser.y
@@ -89,7 +89,7 @@
 %type <cmd> script_init script
 %type <cmd> grubcmd ifclause ifcmd forcmd whilecmd untilcmd
 %type <cmd> command commands1 statement
 
-%pure-parser
+%define api.pure
 %lex-param   { struct grub_parser_param *state };
 %parse-param { struct grub_parser_param *state };
 
