--- a/Makefile.in
+++ b/Makefile.in
@@ -16104,7 +16104,8 @@
 # For libgrub.a
 libgrub.pp: config-util.h grub_script.tab.h grub_script.yy.h $(libgrubmods_a_SOURCES) $(libgrubkern_a_SOURCES)
 	$(CPP) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(libgrubmods_a_CPPFLAGS) $(libgrubkern_a_CPPFLAGS) $(CPPFLAGS) \
-	  -D'GRUB_MOD_INIT(x)=@MARKER@x@' $^ > $@ || (rm -f $@; exit 1)
+	  -include config-util.h -D'GRUB_MOD_INIT(x)=@MARKER@x@' \
+	  $(filter-out config-util.h,$^) > $@ || (rm -f $@; exit 1)
 
 libgrub_a_init.lst: libgrub.pp
 	cat $< | grep '^@MARKER@' | sed 's/@MARKER@\(.*\)@/\1/g' | sort -u > $@ || (rm -f $@; exit 1)
@@ -16115,7 +16116,8 @@
 # For grub-fstest
 grub_fstest.pp: config-util.h $(grub_fstest_SOURCES)
 	$(CPP) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(grub_fstest_CPPFLAGS) $(CPPFLAGS) \
-	  -D'GRUB_MOD_INIT(x)=@MARKER@x@' $^ > $@ || (rm -f $@; exit 1)
+	  -include config-util.h -D'GRUB_MOD_INIT(x)=@MARKER@x@' \
+	  $(filter-out config-util.h,$^) > $@ || (rm -f $@; exit 1)
 
 grub_fstest_init.lst: libgrub.pp grub_fstest.pp
 	cat $^ | grep '^@MARKER@' | sed 's/@MARKER@\(.*\)@/\1/g' | sort -u > $@ || (rm -f $@; exit 1)
--- a/grub-core/Makefile.in
+++ b/grub-core/Makefile.in
@@ -57784,7 +57784,7 @@
 @COND_ENABLE_EFIEMU_TRUE@efiemu64.o: efiemu64_c.o efiemu64_s.o $(TARGET_OBJ2ELEF)
 @COND_ENABLE_EFIEMU_TRUE@	-rm -f $@
 @COND_ENABLE_EFIEMU_TRUE@	-rm -f $@.bin
-@COND_ENABLE_EFIEMU_TRUE@	$(TARGET_CC) -m64 $(EFIEMU64_LINK_FORMAT) -nostdlib -static -Wl,-r -o $@.bin $^
+@COND_ENABLE_EFIEMU_TRUE@	$(TARGET_CC) -m64 $(EFIEMU64_LINK_FORMAT) -nostdlib -static -no-pie -Wl,-r -o $@.bin $^
 @COND_ENABLE_EFIEMU_TRUE@	if test "x$(EFIEMU64_LINK_FORMAT)" = x-arch,x86_64; then \
 @COND_ENABLE_EFIEMU_TRUE@	  $(TARGET_OBJCONV) -felf64 -nu -nd $@.bin $@ || exit 1; \
 @COND_ENABLE_EFIEMU_TRUE@	  rm -f $@.bin; \
