diff --git a/acinclude.m4 b/acinclude.m4
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -233,7 +233,7 @@ AC_CACHE_VAL(grub_cv_check_uscore_uscore_bss_start_symbol,
 void __main (void) {}
 int main (void);
 ]],
-		[[asm ("incl __bss_start")]])],
+		[[asm ("call __bss_start")]])],
 		[grub_cv_check_uscore_uscore_bss_start_symbol=yes],
 		[grub_cv_check_uscore_uscore_bss_start_symbol=no])])
 
@@ -246,7 +246,7 @@ AC_MSG_CHECKING([if edata is defined by the compiler])
 void __main (void);
 void __main (void) {}
 int main (void);]],
-		[[asm ("incl edata")]])],
+		[[asm ("call edata")]])],
 		[grub_cv_check_edata_symbol=yes],
 		[grub_cv_check_edata_symbol=no])])
 
@@ -259,7 +259,7 @@ AC_MSG_CHECKING([if _edata is defined by the compiler])
 void __main (void);
 void __main (void) {}
 int main (void);]],
-		[[asm ("incl _edata")]])],
+		[[asm ("call _edata")]])],
 		[grub_cv_check_uscore_edata_symbol=yes],
 		[grub_cv_check_uscore_edata_symbol=no])])
 
@@ -287,7 +287,7 @@ AC_MSG_CHECKING([if end is defined by the compiler])
 void __main (void);
 void __main (void) {}
 int main (void);]],
-		[[asm ("incl end")]])],
+		[[asm ("call end")]])],
 		[grub_cv_check_end_symbol=yes],
 		[grub_cv_check_end_symbol=no])])
 
@@ -300,7 +300,7 @@ AC_MSG_CHECKING([if _end is defined by the compiler])
 void __main (void);
 void __main (void) {}
 int main (void);]],
-		[[asm ("incl _end")]])],
+		[[asm ("call _end")]])],
 		[grub_cv_check_uscore_end_symbol=yes],
 		[grub_cv_check_uscore_end_symbol=no])])
 
