diff -urN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2018-12-05 15:42:43.093860419 -0600
+++ b/CMakeLists.txt	2018-12-05 15:47:12.917859390 -0600
@@ -41,7 +41,6 @@
 find_package(Qt5 ${QT_MINIMUM_VERSION} CONFIG REQUIRED Widgets DBus X11Extras LinguistTools)
 find_package(Qt5Xdg ${QTXDG_MINIMUM_VERSION} REQUIRED)
 find_package(KF5WindowSystem ${KF5_MINIMUM_VERSION} REQUIRED)
-find_package(PolkitQt5-1 REQUIRED)
 find_package(X11 REQUIRED)
 message(STATUS "Building ${PROJECT_NAME} with Qt ${Qt5Core_VERSION}")
 
@@ -154,13 +153,6 @@
     configdialog/lxqtconfigdialog.ui
 )
 
-set(POLKIT_FILES
-    "${CMAKE_CURRENT_BINARY_DIR}/org.lxqt.backlight.pkexec.policy"
-)
-
-# Build Polkit file
-configure_file ( polkit/org.lxqt.backlight.pkexec.policy.in "${CMAKE_CURRENT_BINARY_DIR}/org.lxqt.backlight.pkexec.policy" )
-
 file(GLOB LXQT_CONFIG_FILES resources/*.conf)
 
 QT5_ADD_DBUS_INTERFACE(DBUS_INTERFACE_SRCS
@@ -363,7 +355,6 @@
     COMPONENT Runtime
 )
 
-install(FILES ${POLKIT_FILES} DESTINATION "${POLKITQT-1_POLICY_FILES_INSTALL_DIR}")
 
 #************************************************
 # Create and install pkgconfig file
