diff -urN a/Makefile.pre.in b/Makefile.pre.in
--- a/Makefile.pre.in	2019-04-21 15:27:16.398807108 -0500
+++ b/Makefile.pre.in	2019-04-21 16:12:23.336934552 -0500
@@ -1258,79 +1258,21 @@
 
 # Install the library
 XMLLIBSUBDIRS=  xml xml/dom xml/etree xml/parsers xml/sax
-LIBSUBDIRS=	tkinter tkinter/test tkinter/test/test_tkinter \
-		tkinter/test/test_ttk site-packages test \
-		test/audiodata \
-		test/capath test/data \
-		test/cjkencodings test/decimaltestdata test/xmltestdata \
-		test/dtracedata \
-		test/eintrdata \
-		test/imghdrdata \
-		test/libregrtest \
-		test/subprocessdata test/sndhdrdata test/support \
-		test/tracedmodules test/encoded_modules \
-		test/test_import \
-		test/test_import/data \
-		test/test_import/data/circular_imports \
-		test/test_import/data/circular_imports/subpkg \
-		test/test_import/data/package \
-		test/test_import/data/package2 \
+LIBSUBDIRS=	site-packages \
 		importlib \
-		test/test_importlib \
-		test/test_importlib/builtin \
-		test/test_importlib/data01 \
-		test/test_importlib/data01/subdirectory \
-		test/test_importlib/data02 \
-		test/test_importlib/data02/one \
-		test/test_importlib/data02/two \
-		test/test_importlib/data03 \
-		test/test_importlib/data03/namespace \
-		test/test_importlib/data03/namespace/portion1 \
-		test/test_importlib/data03/namespace/portion2 \
-		test/test_importlib/extension \
-		test/test_importlib/frozen \
-		test/test_importlib/import_ \
-		test/test_importlib/namespace_pkgs \
-		test/test_importlib/namespace_pkgs/both_portions \
-		test/test_importlib/namespace_pkgs/both_portions/foo \
-		test/test_importlib/namespace_pkgs/module_and_namespace_package \
-		test/test_importlib/namespace_pkgs/module_and_namespace_package/a_test \
-		test/test_importlib/namespace_pkgs/not_a_namespace_pkg \
-		test/test_importlib/namespace_pkgs/not_a_namespace_pkg/foo \
-		test/test_importlib/namespace_pkgs/portion1 \
-		test/test_importlib/namespace_pkgs/portion1/foo \
-		test/test_importlib/namespace_pkgs/portion2 \
-		test/test_importlib/namespace_pkgs/portion2/foo \
-		test/test_importlib/namespace_pkgs/project1 \
-		test/test_importlib/namespace_pkgs/project1/parent \
-		test/test_importlib/namespace_pkgs/project1/parent/child \
-		test/test_importlib/namespace_pkgs/project2 \
-		test/test_importlib/namespace_pkgs/project2/parent \
-		test/test_importlib/namespace_pkgs/project2/parent/child \
-		test/test_importlib/namespace_pkgs/project3 \
-		test/test_importlib/namespace_pkgs/project3/parent \
-		test/test_importlib/namespace_pkgs/project3/parent/child \
-		test/test_importlib/source \
-		test/test_importlib/zipdata01 \
-		test/test_importlib/zipdata02 \
 		asyncio \
-		test/test_asyncio \
 		collections concurrent concurrent/futures encodings \
-		email email/mime test/test_email test/test_email/data \
+		email email/mime \
 		ensurepip ensurepip/_bundled \
-		html json test/test_json http dbm xmlrpc \
-		sqlite3 sqlite3/test \
+		html json http dbm xmlrpc \
 		logging csv wsgiref urllib \
-		lib2to3 lib2to3/fixes lib2to3/pgen2 lib2to3/tests \
-		lib2to3/tests/data lib2to3/tests/data/fixers \
-		lib2to3/tests/data/fixers/myfixes \
+		lib2to3 lib2to3/fixes lib2to3/pgen2 \
 		ctypes ctypes/test ctypes/macholib \
 		idlelib idlelib/Icons idlelib/idle_test \
 		distutils distutils/command distutils/tests $(XMLLIBSUBDIRS) \
-		test/test_tools test/test_warnings test/test_warnings/data \
 		turtledemo \
 		multiprocessing multiprocessing/dummy \
-		unittest unittest/test unittest/test/testmock \
+		unittest \
 		venv venv/scripts venv/scripts/common venv/scripts/posix \
 		curses pydoc_data
 libinstall:	build_all $(srcdir)/Modules/xxmodule.c
diff -urN a/Modules/Setup.dist b/Modules/Setup.dist
--- a/Modules/Setup.dist	2019-04-21 15:27:16.450807344 -0500
+++ b/Modules/Setup.dist	2019-04-21 15:28:21.899106070 -0500
@@ -363,7 +363,7 @@
 # Uncommenting the following line tells makesetup that all following modules
 # are not built (see above for more detail).
 #
-#*disabled*
-#
-#_sqlite3 _tkinter _curses pyexpat
-#_codecs_jp _codecs_kr _codecs_tw unicodedata
+*disabled*
+
+ossaudiodev _sqlite3 _tkinter _curses _codecs_cn _codecs_hk _codecs_iso2022 _codecs_jp _codecs_kr _codecs_tw
+
