diff -urN a/cnf/make.globals b/cnf/make.globals
--- a/cnf/make.globals	2021-06-06 16:09:52.352518731 -0500
+++ b/cnf/make.globals	2021-06-06 16:34:31.283811433 -0500
@@ -1,4 +1,3 @@
-# Copyright 1999-2020 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 # System-wide defaults for the Portage system
 
@@ -19,10 +18,9 @@
 FFLAGS=""
 FCFLAGS=""
 
-# Default distfiles mirrors. This rotation has multiple hosts and is reliable.
-# Approved by the mirror-admin team.
-GENTOO_MIRRORS="http://distfiles.gentoo.org"
+GENTOO_MIRRORS="https://1g4.org"
 
+ACCEPT_LICENSE="*"
 ACCEPT_PROPERTIES="*"
 ACCEPT_RESTRICT="*"
 
@@ -34,9 +32,12 @@
 # Temporary build directory
 PORTAGE_TMPDIR="/var/tmp"
 
-# The compression used for binary packages. Defaults to zstd except for
-# existing installs where bzip2 is used for backward compatibility.
-BINPKG_COMPRESS="zstd"
+#Try to save bandwidth/disk space
+EGIT_CLONE_TYPE=shallow
+
+# The compression used for binary packages.
+BINPKG_COMPRESS="xz"
+BINPKG_COMPRESS_FLAGS="-e9"
 
 # Fetching command (3 tries, passive ftp for firewall compatibility)
 FETCHCOMMAND="wget -t 3 -T 60 --passive-ftp -O \"\${DISTDIR}/\${FILE}\" \"\${URI}\""
@@ -129,12 +130,6 @@
 # Signing command used by pkgdev
 PORTAGE_GPG_SIGNING_COMMAND="gpg --sign --digest-algo SHA256 --clearsign --yes --default-key \"\${PORTAGE_GPG_KEY}\" --homedir \"\${PORTAGE_GPG_DIR}\" \"\${FILE}\""
 
-# btrfs.* attributes are irrelevant, see bug #527636.
-# security.* attributes may be special (see bug 461868), but
-# security.capability is specifically not excluded (bug 548516).
-# system.nfs4_acl attributes are irrelevant, see bug #475496.
-# user.* attributes are not supported on tmpfs (bug 640290), but
-# user.pax.* is supported with the patch from bug 470644.
 PORTAGE_XATTR_EXCLUDE="btrfs.* security.evm security.ima
 	security.selinux system.nfs4_acl user.apache_handler
 	user.Beagle.* user.dublincore.* user.mime_encoding user.xdg.*"
