diff -ur a/bin/phase-helpers.sh b/bin/phase-helpers.sh
--- a/bin/phase-helpers.sh	2025-03-10 07:43:07.862035231 -0500
+++ b/bin/phase-helpers.sh	2025-03-10 07:48:45.293166781 -0500
@@ -629,7 +629,11 @@
 			--infodir="${EPREFIX}"/usr/share/info \
 			--datadir="${EPREFIX}"/usr/share \
 			--sysconfdir="${EPREFIX}"/etc \
-			--localstatedir="${EPREFIX}"/var/lib \
+			--localstatedir="${EPREFIX}"/var \
+			--libdir="${EPREFIX}"/usr/lib \
+			--bindir="${EPREFIX}"/usr/bin \
+			--sbindir="${EPREFIX}"/usr/bin \
+			--libexecdir="${EPREFIX}"/usr/libexec \
 			"${conf_args[@]}" \
 			"$@" \
 			"${EXTRA_ECONF[@]}"
@@ -682,7 +686,7 @@
 			${MAKE:-make} -n prefix="${ED%/}/usr" \
 				datadir="${ED%/}/usr/share" \
 				infodir="${ED%/}/usr/share/info" \
-				localstatedir="${ED%/}/var/lib" \
+				localstatedir="${ED%/}/var" \
 				mandir="${ED%/}/usr/share/man" \
 				sysconfdir="${ED%/}/etc" \
 				${LOCAL_EXTRA_EINSTALL} \
@@ -693,7 +697,7 @@
 		if ! ${MAKE:-make} prefix="${ED%/}/usr" \
 			datadir="${ED%/}/usr/share" \
 			infodir="${ED%/}/usr/share/info" \
-			localstatedir="${ED%/}/var/lib" \
+			localstatedir="${ED%/}/var" \
 			mandir="${ED%/}/usr/share/man" \
 			sysconfdir="${ED%/}/etc" \
 			${LOCAL_EXTRA_EINSTALL} \
@@ -777,18 +781,6 @@
 	if [[ -f Makefile || -f GNUmakefile || -f makefile ]] ; then
 		emake DESTDIR="${D}" install
 	fi
-
-	if ! declare -p DOCS &>/dev/null ; then
-		local d
-		for d in README* ChangeLog AUTHORS NEWS TODO CHANGES \
-				THANKS BUGS FAQ CREDITS CHANGELOG ; do
-			[[ -s "${d}" ]] && dodoc "${d}"
-		done
-	elif ___is_indexed_array_var DOCS ; then
-		dodoc "${DOCS[@]}"
-	else
-		dodoc ${DOCS}
-	fi
 }
 
 __eapi6_src_prepare() {
@@ -806,7 +798,8 @@
 		emake DESTDIR="${D}" install
 	fi
 
-	einstalldocs
+	cleanup_install
+	dedup_symlink "${D}"
 }
 
 __eapi8_src_prepare() {
