diff -urN a/bin/phase-helpers.sh b/bin/phase-helpers.sh
--- a/bin/phase-helpers.sh	2025-06-09 07:54:26.511821273 -0400
+++ b/bin/phase-helpers.sh	2025-06-23 12:48:59.693825114 -0400
@@ -639,7 +639,12 @@
 			--infodir="${EPREFIX}"/usr/share/info \
 			--datadir="${EPREFIX}"/usr/share \
 			--sysconfdir="${EPREFIX}"/etc \
-			--localstatedir="${EPREFIX}"/var/lib \
+			--sharedstatedir="${EPREFIX}"/var/lib \
+			--localstatedir="${EPREFIX}"/var \
+			--libdir="${EPREFIX}"/usr/lib \
+			--bindir="${EPREFIX}"/usr/bin \
+			--sbindir="${EPREFIX}"/usr/bin \
+			--libexecdir="${EPREFIX}"/usr/libexec \
 			"${conf_args[@]}" \
 			"$@" \
 			"${EXTRA_ECONF[@]}"
@@ -692,9 +697,14 @@
 			${MAKE:-make} -n prefix="${ED%/}/usr" \
 				datadir="${ED%/}/usr/share" \
 				infodir="${ED%/}/usr/share/info" \
-				localstatedir="${ED%/}/var/lib" \
+				sharedstatedir="${ED%/}/var/lib" \
+				localstatedir="${ED%/}/var" \
 				mandir="${ED%/}/usr/share/man" \
 				sysconfdir="${ED%/}/etc" \
+				libdir="${ED%/}/usr/lib" \
+				bindir="${ED%/}/usr/bin" \
+				sbindir="${ED%/}/usr/bin" \
+				libexecdir="${ED%/}/usr/libexec" \
 				${LOCAL_EXTRA_EINSTALL} \
 				${MAKEOPTS} -j1 \
 				"$@" ${EXTRA_EMAKE} install
@@ -703,9 +713,14 @@
 		if ! ${MAKE:-make} prefix="${ED%/}/usr" \
 			datadir="${ED%/}/usr/share" \
 			infodir="${ED%/}/usr/share/info" \
-			localstatedir="${ED%/}/var/lib" \
+			sharedstatedir="${ED%/}/var/lib" \
+			localstatedir="${ED%/}/var" \
 			mandir="${ED%/}/usr/share/man" \
 			sysconfdir="${ED%/}/etc" \
+			libdir="${ED%/}/usr/lib" \
+			bindir="${ED%/}/usr/bin" \
+			sbindir="${ED%/}/usr/bin" \
+			libexecdir="${ED%/}/usr/libexec" \
 			${LOCAL_EXTRA_EINSTALL} \
 			${MAKEOPTS} -j1 \
 			"$@" ${EXTRA_EMAKE} install
@@ -788,21 +803,8 @@
 		emake DESTDIR="${D}" install
 	fi
 
-	# To use declare -p determines whether a variable was declared but not
-	# whether it was set. Unfortunately, the language of EAPI 4 requires
-	# that it be this way.
-	# https://projects.gentoo.org/pms/4/pms.html#x1-10400010.1.9
-	if ! declare -p DOCS &>/dev/null ; then
-		local d
-		for d in README* ChangeLog AUTHORS NEWS TODO CHANGES \
-				THANKS BUGS FAQ CREDITS CHANGELOG ; do
-			[[ -s "${d}" ]] && dodoc "${d}"
-		done
-	elif [[ ${DOCS@a} == *a* ]] ; then
-		dodoc "${DOCS[@]}"
-	else
-		dodoc ${DOCS}
-	fi
+	cleanup_install
+	dedup_symlink "${D}"
 }
 
 __eapi6_src_prepare() {
@@ -820,7 +822,8 @@
 		emake DESTDIR="${D}" install
 	fi
 
-	einstalldocs
+	cleanup_install
+	dedup_symlink "${D}"
 }
 
 __eapi8_src_prepare() {
@@ -954,34 +957,6 @@
 	}
 fi
 
-if ___eapi_has_einstalldocs; then
-	einstalldocs() (
-		local f
-
-		if [[ ! -v DOCS ]]; then
-			for f in README* ChangeLog AUTHORS NEWS TODO CHANGES \
-				THANKS BUGS FAQ CREDITS CHANGELOG
-			do
-				if [[ -f ${f} && -s ${f} ]]; then
-					docinto / && dodoc "${f}"
-				fi
-			done
-		elif [[ ${DOCS@a} == *a* ]] && (( ${#DOCS[@]} )); then
-			docinto / && dodoc -r "${DOCS[@]}"
-		elif [[ ${DOCS@a} != *[aA]* && ${DOCS} ]]; then
-			# shellcheck disable=2086
-			docinto / && dodoc -r ${DOCS}
-		fi
-
-		if [[ ${HTML_DOCS@a} == *a* ]] && (( ${#HTML_DOCS[@]} )); then
-			docinto html && dodoc -r "${HTML_DOCS[@]}"
-		elif [[ ${HTML_DOCS@a} != *[aA]* && ${HTML_DOCS} ]]; then
-			# shellcheck disable=2086
-			docinto html && dodoc -r ${HTML_DOCS}
-		fi
-	)
-fi
-
 if ___eapi_has_eapply; then
 	eapply() {
 		local LC_ALL LC_COLLATE=C f i patch_cmd path
