From 053d49367c51bcdaa97f6f22188e5b62070c566c Mon Sep 17 00:00:00 2001
From: Christoph Schulz <develop@kristov.de>
Date: Sun, 18 Dec 2016 13:54:35 -0500
Subject: [PATCH] * main.c (switches): [SV 48809] Accept obsolete jobserver
 flag.

---
 main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/main.c b/main.c
index 346cc0d..b170e8e 100644
--- a/main.c
+++ b/main.c
@@ -464,6 +464,7 @@ static const struct command_switch switches[] =
       "warn-undefined-variables" },
     { CHAR_MAX+6, strlist, &eval_strings, 1, 0, 0, 0, 0, "eval" },
     { CHAR_MAX+7, string, &sync_mutex, 1, 1, 0, 0, 0, "sync-mutex" },
+    { CHAR_MAX+8, string, &jobserver_auth, 1, 0, 0, 0, 0, "jobserver-fds" },
     { 0, 0, 0, 0, 0, 0, 0, 0, 0 }
   };
 
