From 9c487cc5376b8438d3d4052dedd6bed90f40cd7e Mon Sep 17 00:00:00 2001
From: Jaak Ristioja <jaak@ristioja.ee>
Date: Sun, 18 Dec 2016 09:48:46 -0500
Subject: [PATCH] * expand.c (variable_append): [SV 49113] Possible null ptr
 deref

---
 expand.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/expand.c b/expand.c
index 0b5fd01..60c6071 100644
--- a/expand.c
+++ b/expand.c
@@ -490,13 +490,15 @@ variable_append (const char *name, unsigned int length,
 {
   const struct variable *v;
   char *buf = 0;
-  /* If this set is local and the next is not a parent, then next is local.  */
-  int nextlocal = local && set->next_is_parent == 0;
+  int nextlocal;
 
   /* If there's nothing left to check, return the empty buffer.  */
   if (!set)
     return initialize_variable_output ();
 
+  /* If this set is local and the next is not a parent, then next is local.  */
+  nextlocal = local && set->next_is_parent == 0;
+
   /* Try to find the variable in this variable set.  */
   v = lookup_variable_in_set (name, length, set->set);
 
