diff --git a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -162,7 +162,7 @@
 	( cd $(srcdir) && $(abs_builddir)/flex$(EXEEXT) \
 	  $(AM_LFLAGS) $(LFLAGS) -o scan.c -t scan.l ) >$@
 
-stage2compare: stage1scan.c
+stage2compare: stage1scan.c flex$(EXEEXT)
 	@rm -f stage2scan.c; \
 	$(MAKE) $(AM_MAKEFLAGS) stage2scan.c; \
 	echo Comparing stage1scan.c and stage2scan.c; \
