diff --git a/src/cpp-flex.skl b/src/cpp-flex.skl
--- a/src/cpp-flex.skl
+++ b/src/cpp-flex.skl
@@ -1188,6 +1188,6 @@
-m4_ifdef( [[M4_EXTRA_TYPE_DEFS]],
+m4_ifdef( [[M4_MODE_EXTRA_TYPE]],
 [[
-#define YY_EXTRA_TYPE M4_EXTRA_TYPE_DEFS
+#define YY_EXTRA_TYPE M4_MODE_EXTRA_TYPE
 ]],
 [[
 #ifndef YY_EXTRA_TYPE
diff --git a/tests/yyextra_nr.l b/tests/yyextra_nr.l
--- a/tests/yyextra_nr.l
+++ b/tests/yyextra_nr.l
@@ -29,5 +29,6 @@
 */
 
+#include <assert.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include "config.h"
@@ -54,7 +55,10 @@
 
 %%
 
-.|\r|\n  { append_char (yytext[0],yyscanner); }
+.|\r|\n  {
+    assert(yyextra->curr_len <= yyextra->max_len);
+    append_char (yytext[0],yyscanner);
+}
 
 %%
 
