--- include/crt/host_config.h	2017-12-01 17:36:56.000000000 -0600
+++ include/crt/host_config.h	2018-03-20 23:10:46.848756718 -0500
@@ -116,9 +116,9 @@

 #if defined(__GNUC__)

-#if __GNUC__ > 7
+#if __GNUC__ > 8

-#error -- unsupported GNU version! gcc versions later than 7 are not supported!
+#error -- unsupported GNU version! gcc versions later than 8 are not supported!

 #endif /* __GNUC__ > 7 */

