diff -urN a/lib/portage/_emirrordist/Config.py b/lib/portage/_emirrordist/Config.py
--- a/lib/portage/_emirrordist/Config.py	2020-04-11 18:37:40.078140534 -0500
+++ b/lib/portage/_emirrordist/Config.py	2020-04-11 23:54:49.929607717 -0500
@@ -10,7 +10,6 @@
 
 import portage
 from portage import os
-from portage.package.ebuild.fetch import MirrorLayoutConfig
 from portage.util import grabdict, grablines
 from portage.util._ShelveUnicodeWrapper import ShelveUnicodeWrapper
 
@@ -73,13 +72,6 @@
 			self.deletion_db = self._open_shelve(
 				options.deletion_db, 'deletion')
 
-		self.layout_conf = MirrorLayoutConfig()
-		if options.layout_conf is None:
-			options.layout_conf = os.path.join(self.distfiles,
-					'layout.conf')
-		self.layout_conf.read_from_file(options.layout_conf)
-		self.layouts = self.layout_conf.get_all_layouts()
-
 	def _open_log(self, log_desc, log_path, mode):
 
 		if log_path is None or self.options.dry_run:
diff -urN a/lib/portage/package/ebuild/fetch.py b/lib/portage/package/ebuild/fetch.py
--- a/lib/portage/package/ebuild/fetch.py	2020-04-11 18:37:40.106140712 -0500
+++ b/lib/portage/package/ebuild/fetch.py	2020-04-11 23:53:23.673171957 -0500
@@ -498,26 +498,8 @@
 			pass
 
 	ts, data = cache.get(mirror_url, (0, None))
-	# refresh at least daily
-	if ts >= time.time() - 86400:
-		mirror_conf.deserialize(data)
-	else:
-		tmpfile = '.layout.conf.%s' % urlparse(mirror_url).hostname
-		try:
-			if fetch({tmpfile: (mirror_url + '/distfiles/layout.conf',)},
-					mysettings, force=1, try_mirrors=0):
-				tmpfile = os.path.join(mysettings['DISTDIR'], tmpfile)
-				mirror_conf.read_from_file(tmpfile)
-			else:
-				raise IOError()
-		except (ConfigParserError, IOError, UnicodeDecodeError):
-			pass
-		else:
-			cache[mirror_url] = (time.time(), mirror_conf.serialize())
-			if cache_path is not None:
-				f = atomic_ofstream(cache_path, 'w')
-				json.dump(cache, f)
-				f.close()
+
+	mirror_conf.deserialize(data)
 
 	return (mirror_url + "/distfiles/" +
 			mirror_conf.get_best_supported_layout().get_path(filename))
