--- a/mkfs/rootdir.c	2026-04-02 16:15:45.175401368 -0500
+++ b/mkfs/rootdir.c	2026-04-02 16:15:45.176774011 -0500
@@ -798,7 +798,7 @@
 	bool datasum = true;
 	ssize_t comp_ret;
 	u64 flags = btrfs_stack_inode_flags(btrfs_inode);
-	off64_t next;
+	off_t next;
 
 	if (g_do_reflink || flags & BTRFS_INODE_NOCOMPRESS)
 		do_comp = false;
@@ -808,11 +808,11 @@
 		do_comp = false;
 	}
 
-	next = lseek64(source->fd, file_pos, SEEK_DATA);
+	next = lseek(source->fd, file_pos, SEEK_DATA);
 	/* The current offset is inside a hole to the next of the file. */
-	if (next == (off64_t)-1 && errno == ENXIO)
+	if (next == (off_t)-1 && errno == ENXIO)
 		next = round_up(source->size, sectorsize);
-	if (next != (off64_t)-1 && next > file_pos && IS_ALIGNED(next, sectorsize)) {
+	if (next != (off_t)-1 && next > file_pos && IS_ALIGNED(next, sectorsize)) {
 		/*
 		 * Limit the hole size to 1G, this is to avoid overflow int type,
 		 * or a hole length >= 2G can be treated as an error.
@@ -834,8 +834,8 @@
 	 * We have skippted to the next data, try to locate the next hole
 	 * to limit the read size.
 	 */
-	next = lseek64(source->fd, file_pos, SEEK_HOLE);
-	if (next == (off64_t)-1 || !IS_ALIGNED(next, sectorsize) || next > source->size)
+	next = lseek(source->fd, file_pos, SEEK_HOLE);
+	if (next == (off_t)-1 || !IS_ALIGNED(next, sectorsize) || next > source->size)
 		next = source->size;
 
 	buf_size = do_comp ? BTRFS_MAX_COMPRESSED : MAX_EXTENT_SIZE;
