From b040ac71cd19cb5fb193e8d5c6cd608979be363c Mon Sep 17 00:00:00 2001
From: Tim Harder <radhermit@gmail.com>
Date: Sun, 17 Jun 2018 05:29:14 -0400
Subject: [PATCH] ebd/helpers/common/prepstrip: fix compat with >=file-5.33

---
 ebd/helpers/common/prepstrip | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/ebd/helpers/common/prepstrip b/ebd/helpers/common/prepstrip
index c5cedaf40..0eb5af796 100755
--- a/ebd/helpers/common/prepstrip
+++ b/ebd/helpers/common/prepstrip
@@ -67,7 +67,8 @@ for x in $(scanelf -yqRBF '#k%F' -k '.symtab' "$@"); do
 	# actually causes problems.  install sources for all
 	# elf types though cause that stuff is good.
 
-	if [[ ${f} == *"SB executable"* || ${f} == *"SB shared object"* ]]; then
+	if [[ ${f} == *"SB executable"* || ${f} == *"SB pie executable"* ||
+			${f} == *"SB shared object"* ]]; then
 		save_elf_sources "${x}"
 	elif [[ ${f} == *"SB relocatable"* ]]; then
 		save_elf_sources "${x}"
