diff --git a/lib/config/defaults.h b/lib/config/defaults.h
--- a/lib/config/defaults.h
+++ b/lib/config/defaults.h
@@ -67,7 +67,11 @@
 #define DEFAULT_METADATA_READ_ONLY 0
 #define DEFAULT_LVDISPLAY_SHOWS_FULL_DEVICE_PATH 0
 #define DEFAULT_UNKNOWN_DEVICE_NAME "[unknown]"
+#if defined(HAVE_LIBAIO_H) && !defined(LVM_NO_LIBAIO)
 #define DEFAULT_USE_AIO 1
+#else
+#define DEFAULT_USE_AIO 0
+#endif
 
 #define DEFAULT_SANLOCK_LV_EXTEND_MB 256
 #define DEFAULT_SANLOCK_ALIGN_SIZE 8 /* in MiB, applies to 4K disks only */
diff --git a/lib/device/bcache.c b/lib/device/bcache.c
--- a/lib/device/bcache.c
+++ b/lib/device/bcache.c
@@ -27,7 +27,9 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <stdint.h>
+#if defined(HAVE_LIBAIO_H) && !defined(LVM_NO_LIBAIO)
 #include <libaio.h>
+#endif
 #include <unistd.h>
 #include <linux/fs.h>
 #include <sys/user.h>
@@ -38,6 +40,10 @@
 static int _fd_table_size = 0;
 static int *_fd_table = NULL;
 
+static int _last_byte_di;
+static uint64_t _last_byte_offset;
+static size_t _last_byte_sector_size;
+
 static inline int _di_valid(int di)
 {
 	if (di < 0 || di >= _fd_table_size) {
@@ -68,6 +74,8 @@
 
 //----------------------------------------------------------------
 
+#if defined(HAVE_LIBAIO_H) && !defined(LVM_NO_LIBAIO)
+
 struct control_block {
 	struct dm_list list;
 	void *context;
@@ -176,10 +184,6 @@
 	free(e);
 }
 
-static int _last_byte_di;
-static uint64_t _last_byte_offset;
-static size_t _last_byte_sector_size;
-
 static bool _async_issue(struct io_engine *ioe, enum dir d, int di,
 			 sector_t sb, sector_t se, void *data, void *context)
 {
@@ -428,6 +432,15 @@
 	return &e->e;
 }
 
+#else
+
+struct io_engine *create_async_io_engine(void)
+{
+	return NULL;
+}
+
+#endif
+
 //----------------------------------------------------------------
 
 struct sync_io {
