diff --git a/lib/config/defaults.h b/lib/config/defaults.h
index 0e0f178..eef9575 100644
--- a/lib/config/defaults.h
+++ b/lib/config/defaults.h
@@ -68,7 +68,11 @@
 #define DEFAULT_METADATA_READ_ONLY 0
 #define DEFAULT_LVDISPLAY_SHOWS_FULL_DEVICE_PATH 0
 #define DEFAULT_UNKNOWN_DEVICE_NAME "[unknown]"
+#if defined(HAVE_LIBAIO_H) && !defined(LVM_NO_LIBAIO)
 #define DEFAULT_USE_AIO 1
+#else
+#define DEFAULT_USE_AIO 0
+#endif
 
 #define DEFAULT_SANLOCK_LV_EXTEND_MB 256
 #define DEFAULT_SANLOCK_ALIGN_SIZE 8 /* in MiB, applies to 4K disks only */
diff --git a/lib/device/bcache.c b/lib/device/bcache.c
index e1dbe0e..6017f4b 100644
--- a/lib/device/bcache.c
+++ b/lib/device/bcache.c
@@ -27,7 +27,9 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <stdint.h>
+#if defined(HAVE_LIBAIO_H) && !defined(LVM_NO_LIBAIO)
 #include <libaio.h>
+#endif
 #include <unistd.h>
 #include <linux/fs.h>
 #include <sys/user.h>
@@ -38,6 +40,9 @@
 static int _fd_table_size = 0;
 static int *_fd_table = NULL;
 
+static int _last_byte_di;
+static uint64_t _last_byte_offset;
+static int _last_byte_sector_size;
 
 //----------------------------------------------------------------
 
@@ -58,6 +63,8 @@ static inline struct dm_list *_list_pop(struct dm_list *head)
 
 //----------------------------------------------------------------
 
+#if defined(HAVE_LIBAIO_H) && !defined(LVM_NO_LIBAIO)
+
 struct control_block {
 	struct dm_list list;
 	void *context;
@@ -166,10 +173,6 @@ static void _async_destroy(struct io_engine *ioe)
 	free(e);
 }
 
-static int _last_byte_di;
-static uint64_t _last_byte_offset;
-static int _last_byte_sector_size;
-
 static bool _async_issue(struct io_engine *ioe, enum dir d, int di,
 			 sector_t sb, sector_t se, void *data, void *context)
 {
@@ -418,6 +421,15 @@ struct io_engine *create_async_io_engine(void)
 	return &e->e;
 }
 
+#else
+
+struct io_engine *create_async_io_engine(void)
+{
+	return NULL;
+}
+
+#endif
+
 //----------------------------------------------------------------
 
 struct sync_io {
