diff -urN a/library/escape.c b/library/escape.c
--- a/library/escape.c
+++ b/library/escape.c
@@ -21,6 +21,7 @@
 #include <limits.h>
 #include <stdio.h>
 #include <string.h>
+#include <langinfo.h>
 
 #include "escape.h"
 #include "readproc.h"
diff -urN a/src/w.c b/src/w.c
--- a/src/w.c
+++ b/src/w.c
@@ -43,11 +43,7 @@
 #include <termios.h>
 #include <time.h>
 #include <unistd.h>
-#ifdef HAVE_UTMPX_H
-#	include <utmpx.h>
-#else
-#	include <utmp.h>
-#endif
+#include <utmp.h>
 #include <arpa/inet.h>
 
 #include "c.h"
