diff -urN a/configure.ac b/configure.ac
--- a/configure.ac	2020-04-10 21:56:15.922538155 -0500
+++ b/configure.ac	2020-04-10 21:57:48.187118601 -0500
@@ -457,22 +457,11 @@
 
 
 #
-# Extra features
-#
-build_doc=yes
-AC_ARG_ENABLE([doc], AC_HELP_STRING([--disable-doc],
-                                    [do not build the documentation]),
-                     build_doc=$enableval, build_doc=yes)
-AM_CONDITIONAL([BUILD_DOC], [test "x$build_doc" != xno])
-
-
-#
 # Create the config files.
 #
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([m4/Makefile])
 AC_CONFIG_FILES([src/Makefile])
-AC_CONFIG_FILES([doc/Makefile])
 AC_CONFIG_FILES([tests/Makefile])
 AC_CONFIG_FILES([src/libassuan-config], [chmod +x src/libassuan-config])
 AC_CONFIG_FILES([src/versioninfo.rc])
diff -urN a/Makefile.am b/Makefile.am
--- a/Makefile.am	2020-04-10 21:56:15.914538100 -0500
+++ b/Makefile.am	2020-04-10 21:57:27.735004820 -0500
@@ -20,23 +20,17 @@
 
 
 ACLOCAL_AMFLAGS = -I m4
-DISTCHECK_CONFIGURE_FLAGS = --enable-doc
 
 # (A suitable gitlog-to-changelog script can be found in GnuPG master.)
 GITLOG_TO_CHANGELOG=gitlog-to-changelog
 
 EXTRA_DIST = autogen.sh autogen.rc README.GIT VERSION                   \
-             ChangeLog-2011 doc/ChangeLog-2011 src/ChangeLog-2011 	\
+             ChangeLog-2011 src/ChangeLog-2011 	\
              tests/ChangeLog-2011 contrib/ChangeLog-2011     		\
              build-aux/git-log-footer build-aux/git-log-fix
 
-if BUILD_DOC
-doc = doc
-else
-doc =
-endif
 
-SUBDIRS = m4 src $(doc) tests
+SUBDIRS = m4 src tests
 
 
 dist-hook: gen-ChangeLog
