--- a/mysys_ssl/openssl.c
+++ b/mysys_ssl/openssl.c
@@ -39,4 +39,4 @@
 static void *coc_malloc(size_t size
-#ifndef LIBRESSL_VERSION_NUMBER
+#if !defined(LIBRESSL_VERSION_NUMBER) || defined(VESK_VERSION_NUMBER)
                         , const char *f __attribute__((unused)),
                         int l __attribute__((unused))
@@ -54,4 +54,4 @@
 static void *coc_realloc(void *addr, size_t num
-#ifndef LIBRESSL_VERSION_NUMBER
+#if !defined(LIBRESSL_VERSION_NUMBER) || defined(VESK_VERSION_NUMBER)
                          , const char *file __attribute__((unused)),
                          int line __attribute__((unused))
@@ -64,4 +64,4 @@
 static void coc_free(void *addr
-#ifndef LIBRESSL_VERSION_NUMBER
+#if !defined(LIBRESSL_VERSION_NUMBER) || defined(VESK_VERSION_NUMBER)
                      , const char *file __attribute__((unused)),
                      int line __attribute__((unused))
