diff --git a/runtime/worker.rs b/runtime/worker.rs
index b70aa0f9f..ce783108d 100644
--- a/runtime/worker.rs
+++ b/runtime/worker.rs
@@ -78,9 +78,12 @@ pub(crate) static SIGUSR2_RX: LazyLock<tokio::sync::watch::Receiver<()>> =
       loop {
         sigusr2.recv().await;
 
-        // SAFETY: calling into libc, nothing relevant on the Rust side.
-        unsafe {
-          libc::malloc_trim(0);
+        #[cfg(target_env = "gnu")]
+        {
+          // SAFETY: calling into libc, nothing relevant on the Rust side.
+          unsafe {
+            libc::malloc_trim(0);
+          }
         }
 
         if tx.send(()).is_err() {
