From c445d99d4f8aae6932a5385b8fbfb77a77acbff5 Mon Sep 17 00:00:00 2001
From: Sergey Poznyakoff <gray@gnu.org>
Date: Thu, 28 Mar 2019 14:01:58 +0200
Subject: Minor change * scripts/backup.in: Fix typo

---
 scripts/backup.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/scripts/backup.in b/scripts/backup.in
index 1093748..027e82d 100644
--- a/scripts/backup.in
+++ b/scripts/backup.in
@@ -1,7 +1,7 @@
 #! /bin/sh
 # Make backups.
 
-# Copyright 2004-2006, 2013 Free Software Foundation
+# Copyright 2004-2006, 2013, 2019 Free Software Foundation
 
 # This file is part of GNU tar.
 
@@ -149,7 +149,7 @@ message 20 "Variables:"
 message 20 "BACKUP_DIRS=$BACKUP_DIRS"
 message 20 "BACKUP_FILES=$BACKUP_FILES"
 
-# The buch of commands below is run in a subshell for which all output is
+# The bunch of commands below is run in a subshell for which all output is
 # piped through 'tee' to the logfile.  Doing this, instead of having
 # multiple pipelines all over the place, is cleaner and allows access to
 # the exit value from various commands more easily.
-- 
cgit v1.0-41-gc330

